tmail-backend/jmap/extensions/src/main/java/com/linagora/tmail/james/jmap/StandaloneEventAttendanceRepository.java

```diff
@@ -0,0 +1,124 @@
+        this.messageIdManager = messageIdManager;
+        this.sessionProvider = sessionProvider;
+    }
+
+    @Override
+    public Publisher<AttendanceStatus> getAttendanceStatus(Username username, MessageId messageId) {
+        MailboxSession systemMailboxSession = sessionProvider.createSystemSession(username);
+
+        return getFlags(messageId, systemMailboxSession)
+            .map(this::getAttendanceStatusFromFlags);
+    }
+
+    @Override
+    public Publisher<Void> setAttendanceStatus(Username username, MessageId messageId,
+                                               AttendanceStatus attendanceStatus) {
+        MailboxSession systemMailboxSession = sessionProvider.createSystemSession(username);
+
+        return getFlags(messageId, systemMailboxSession)
+            .flatMap(this::removeEventAttendanceFlags)
+            .flatMap(flagsWithoutEventAttendanceFlags -> {
+                Flags flagsToSet = new Flags(flagsWithoutEventAttendanceFlags);
+                getFlagFromAttendanceStatus(attendanceStatus)
+                    .ifPresent(flagsToSet::add);
+                return doSetFlags(messageId, flagsToSet, systemMailboxSession);
+            })
+            .then();
+    }
+
+    private Mono<Flags> removeEventAttendanceFlags(Flags flags) {
+        return Mono.fromSupplier(() -> {
+            Flags flagsWithoutEventAttendanceFlags = new Flags();
+            Arrays.stream(flags.getSystemFlags()).forEach(flagsWithoutEventAttendanceFlags::add);
+            Arrays.stream(flags.getUserFlags())
+                .filter(Predicate.not(EVENT_ATTENDANCE_FLAGS::contains))
+                .forEach(flagsWithoutEventAttendanceFlags::add);
+            return flagsWithoutEventAttendanceFlags;
+        });
+    }
+
+    private Optional<String> getFlagFromAttendanceStatus(AttendanceStatus attendanceStatus) {
+        return switch (attendanceStatus) {
+            case Accepted -> Optional.of("$accepted");
+            case Declined -> Optional.of("$rejected");
+            case Tentative -> Optional.of("$tentativelyaccepted");
+            case Delegated, NeedsAction -> Optional.empty();
+        };
+    }
+
+    private AttendanceStatus getAttendanceStatusFromFlags(Flags flags) {
+        long eventAttendanceFlagsCount = Arrays.stream(flags.getUserFlags())
+            .filter(EVENT_ATTENDANCE_FLAGS::contains)
+            .count();
+
+        if (eventAttendanceFlagsCount > 1) {
+            LOGGER.warn("A message should not have more than one event attendance flag");
+            return AttendanceStatus.NeedsAction;
+        }
+
+        if (flags.contains("$accepted")) {
+            return AttendanceStatus.Accepted;
+        } else if (flags.contains("$rejected")) {
+            return AttendanceStatus.Declined;
+        } else if (flags.contains("$tentativelyaccepted")) {
+            return AttendanceStatus.Tentative;
+        } else {
+            return AttendanceStatus.NeedsAction;
```

Comment:
This is implicit so the day we add an entry into EVENT_ATTENDANCE_FLAGS then we will return NeedsAction which would be a lie.

Reply:
note: That would require that we introduce a new flag that does not exist on tmail-flutter.